include/realtime_tools/mutex.hpp

```diff
@@ -0,0 +1,186 @@
+// Copyright 2024 PAL Robotics S.L.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// \author Sai Kishor Kothakota
+
+#ifndef REALTIME_TOOLS__MUTEX_HPP_
+#define REALTIME_TOOLS__MUTEX_HPP_
+
+#include <errno.h>
+#include <pthread.h>
+#include <cstring>
+#include <iostream>
+#include <stdexcept>
+#include <string>
+
+/**
+ * @brief A pthread mutex wrapper that provides a mutex with the priority inheritance
+ * protocol and a priority ceiling of 99.
+ * The mutex is also error checked and robust.
+ * This mutex is intended to be used in real-time contexts.
+ * @note This mutex is not recursive.
+ */
+namespace realtime_tools
+{
+namespace priority_inheritance
+{
+/**
+ * @brief A class template that provides a pthread mutex with the priority inheritance protocol
+ *
+ * @tparam MutexType The type of the mutex. It can be one of the following: PTHREAD_MUTEX_NORMAL, PTHREAD_MUTEX_RECURSIVE, PTHREAD_MUTEX_ERRORCHECK, PTHREAD_MUTEX_DEFAULT
```

Comment:
I agree with you, I will reduce the scope. Thanks for pointing out

Regarding to the `PTHREAD_MUTEX_DEFAULT`, I choose it because of the explanation given here : https://linux.die.net/man/3/pthread_mutexattr_settype

Reply:
@atzaros Can you confirm this is fine